test/libsolidity/syntaxTests/experimental/parsing/forall_free_function_with_sorts.sol

```diff
@@ -0,0 +1,16 @@
+pragma experimental solidity;
+
+class Self: Class1 {}
+class Self: Class2 {}
+
+forall (A: (Class1, Class2), B: Class1)
```

Comment:
For now I kept the parentheses and did not use the dot as a separator. It was just easier to implement that way. I see that the parser has hacked-in support for accepting single parameters without parentheses. I could extend that but I think that what we have here might be good enough until we start working on syntax.

Reply:
Yeah, looks a bit weird, but fine for now :-).